app/controllers/spree/credit_cards_controller.rb

```diff
@@ -74,6 +81,15 @@ def create_customer(token)
       Stripe::Customer.create(email: spree_current_user.email, source: token)
     end
 
+    def verify(card)
+      options = { metadata: { adding_card: true } }
+      Spree::Gateway::StripeSCA.new.verify(card, options)
+    end
```

Comment:
I see, I dont think this is correct but what it really matters is if it works or not. Working here means the subs payments will be correctly charged on the stripe enterprise account without extra SCA auth required (on the subs job) because the setup intent is correctly setup. This is something we need to test and can prove this current approach right or wrong.

I think the setup_intent needs to be created against the enterprise stripe account, not the platform account. In stripe the payment method and the customer are not the same objects at platform and enterprise account, you can see what the credit_card_cloner does. In subscriptions payments we also clone the customer and it's payment method to the stripe enterprise account. In my head the setup intent cannot be copied/cloned but maybe it's possible to clone it. Anyway, I think we will need at least to clone the setup intent to the enterprise account? Maybe we need to add something to the CreditCardCloner?
I may be wrong and your current approach will work, we need to test with cards that require extra SCA auth all the time without setup_intents.